src/common/hist_util.cc

```diff
@@ -303,7 +303,9 @@ void DenseCuts::Init
   }
   CHECK_EQ(summary_array.size(), in_sketchs->size());
   size_t nbytes = WXQSketch::SummaryContainer::CalcMemCost(max_num_bins * kFactor);
-  sreducer.Allreduce(dmlc::BeginPtr(summary_array), nbytes, summary_array.size());
+  // TODO(chenqin): after remove is_bootstrap, hist break rabit assumeption
+  sreducer.Allreduce(dmlc::BeginPtr(summary_array), nbytes, summary_array.size(),
+    nullptr, nullptr, "DenseCuts");
```

Comment:
Just a suggestion for future.  We can count the number of rows for each column, sync the maximum one in all workers.  Then we can remove this "DenseCuts" and use sparse version instead.

Reply:
I agree, I recall there was a bug raised before due to testing dataset have different column size as training dataset.

Reply:
sounds like this should be a follow up pr